app/code/Magento/Review/Model/Review.php

```diff
@@ -221,9 +222,10 @@ public function aggregate()
     public function getEntitySummary($product, $storeId = 0)
     {
         $summaryData = $this->_summaryModFactory->create()->setStoreId($storeId)->load($product->getId());
-        $summary = new \Magento\Framework\DataObject();
-        $summary->setData($summaryData->getData());
-        $product->setRatingSummary($summary);
+        $product->addData([
+            'reviews_count' => $summaryData->getData('reviews_count'),
+            'rating_summary' => $summaryData->getData('rating_summary')
+        ]);
```

Comment:
Any reason to change this at all?

Reply:
@orlangur 
this part of code changed, because observer adding new columns to select.
As result we don't have DataObject in product data